#!/usr/bin/env python3
PACKAGE = "robot_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("arm_speed",    double_t,    0, "Speed of the arm while moving", 0.15,  0.01, 0.5)
gen.add("gripper_width",    double_t,    0, "Width of the width", 0.06,  0.0, 0.081)
gen.add("gripper_speed", double_t, 0, "Speed of the gripper", 0.5 , 0.01 , 1.0)
gen.add("is_real",   bool_t,   0, "True for real robot, False for Gazebo simulation",  True)

gen.add("a",   double_t,   0, "a value for orientation of end effector",  0.0 , 0.0 , 6.28)
gen.add("b",   double_t,   0, "b value for orientation of end effector",  0.0 , 0.0 , 6.28)
gen.add("c",   double_t,   0, "c value for orientation of end effector",  0.0 , 0.0 , 6.28)

gen.add("x",   double_t,   0, "x value for orientation of end effector", 0.33  , -2.0 , 2.0)
gen.add("y",   double_t,   0, "y value for orientation of end effector",  0.0 , -0.8 , 0.8)
gen.add("z",   double_t,   0, "z value for orientation of end effector",  0.088 , 0.0 , 0.6)


gen.add("force_limit", int_t, 0, "Maximum force the robot can utilize", 20, 1, 30)
gen.add("elbow",   double_t,   0, "angle of the elbow",  0.0 , 0.0 , 6.28)





exit(gen.generate(PACKAGE, "controller", "Robot"))